File: Method4/PrimeNumbers.cpp
#include<iostream>
using namespace std;

int isPrime(int num, int i) { // Time Complexity O(sqrt(N))
  if (num < 2) {
    return 0;
  } else if (num == 2) {
    return 1;
  } else if (num % i == 0) {
    return 0;
  } else if (i * i > num) {
    return 1;
  } else {
    return isPrime(num, i + 1);
  }
}

int main() {
  int num;

  cout << "Enter Value of n... ";
  cin >> num;

  if (isPrime(num, 2)) {
    cout << num << " is a Prime Number.";
  } else {
    cout << num << " is not a Prime Number.";
  }
  return 0;
}

File: Method4/PrimeNumbers.py
def isPrime(num, i): # Time Complexity O(sqrt(N))
  if num < 2:
    return False
  elif num == 2:
    return True
  elif num % i == 0:
    return False
  elif i * i > num:
    return True
  else:
    return isPrime(num, i + 1)

num = int(input('Enter Value of n... '))
if isPrime(num, 2):
  print(num, 'is a Prime Number.')
else:
  print(num,'is not a Prime Number.')